src/System.Drawing.Common/tests/GraphicsTests.cs

```diff
@@ -0,0 +1,374 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.Drawing.Text;
+using Xunit;
+
+namespace System.Drawing.Tests
+{
+    public class GraphicsTests
+    {
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void GetHdc_FromHdc_Roundtrips()
+        {
+            using (var bitmap = new Bitmap(10, 10))
+            using (Graphics graphics = Graphics.FromImage(bitmap))
+            {
+                IntPtr hdc = graphics.GetHdc();
+                Assert.NotEqual(IntPtr.Zero, hdc);
+
+                using (Graphics graphicsCopy = Graphics.FromHdc(hdc))
+                {
+                    VerifyGraphics(graphicsCopy, graphicsCopy.VisibleClipBounds);
+                }
+
+                graphics.ReleaseHdc();
```

Comment:
Worse case scenario is we'd get a memory leak. But it also looks like the finalizer properly releases the HDC if it still exists